### DISCUSSION TITLE
Scenario Outline, Nunit Test Names

### DISCUSSION THREAD

**Discussion by infinitejest69** [reqnroll/Reqnroll#214] [Ideas]
This is not really a bug but a little bit of an issue i am experiencing that i could use help with.
i am using reqnroll after switching from specflow.

i am moving my tests in azure devops pipeline from windows to linux agents. ( for Mutiple reasons)
i was using the [vstest ](https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/vstest-v2?view=azure-pipelines )task to automatically slice the tests over Mutiple agents. that task is not available on the linux agents. so i am manually slicing them based on this [document](https://learn.microsoft.com/en-us/azure/devops/pipelines/test/parallel-testing-any-test-runner?view=azure-devops)

my issue seems to be when slicing tests up to run by name the nunit generated name for scenario outlines looks a little bit like the below with brackets and spaces as as opposed to a single string

when running 
`dotnet test --no-build --list-tests`
```
    Pagination("first","Next Page","second","First Page, Previous Page","No Change",null)
    Pagination("second","Previous Page","first","No Change","First Page, Previous Page",null)
    Pagination("first","Last Page","last","First Page, Previous Page","Next Page, Last Page",null)
    Pagination("last","First Page","first","Next Page, Last Page","First Page, Previous Page",null)
    RowsPerPage("50",null)
    RowsPerPage("25",null)
    RowsPerPage("100",null)
```

imagine the above but on a much large scale with over 2000 tests. i need some way to make the scenario outline tests into a nice single string like normal scenarios either in code generation renroll change or some other way . 

the dotnet test  with filter Name= does not like spaces and all the other characters to escape.
open to ideas thought i would post here incase other have solved this issue with scenario outline names or a way to change how they generate

**Comment by gasparnagy** [comment]
This is not something that we can control unfortunately. For scenario outlines (by default, see below) we generate data-driven tests with the `[TestCase]` attribute and the [NUnit visual studio adapter](https://www.nuget.org/packages/NUnit3TestAdapter) generates the names. 

Maybe you can try the NUnit config parameter [DefaultTestNamePattern](https://docs.nunit.org/articles/vs-test-adapter/Tips-And-Tricks.html#defaulttestnamepattern) or some [other config for NUnit in runsettings files](https://docs.nunit.org/articles/vs-test-adapter/Tips-And-Tricks.html).

With Reqnroll the only thing you can do to force generating individual test methods for scenario outline examples by setting the `allowRowTests` to `false` in [config](https://docs.reqnroll.net/latest/installation/configuration.html#generator).

Does that help?

**Reply by infinitejest69** [reply]
Thank you will give that a go
